### PR TITLE
add test for listfile

### PR DESCRIPTION
**Pull request by ysmaoui**
json lint the json output when --listfile is used

### CODE REVIEW

**Comment by ysmaoui** [comment]
it looks like the json output of --listfile has a bug ( an extra comma at the end ) 

I added a test to catch this issue.

still not sure where to fix the formatting.

File: checksec [hunk 1]
```diff
@@ -289,8 +289,11 @@ chk_file_list() {
     echo_message "RELRO           STACK CANARY      NX            PIE             RPATH      RUNPATH\tSymbols\t\tFORTIFY\tFortified\tFortifiable\tFILE\n" '' '' '{\n'
   fi
 
+  LAST_LINE_NUMBER=$(wc -l < "${CHK_FILE_LIST}")
```
Comment (slimm609) [inline checksec:292]: This needs to be updated in the src directory, then generate the final code hack/build.sh

Comment (ysmaoui) [reply checksec:292]: @slimm609  sorry I missed that. Now the changes are made to the src/functions/chk_file.sh

**Comment by slimm609** [comment]
Shellcheck is failing.

**Comment by slimm609** [comment]
@ysmaoui Did you run `make build` locally?  It looks like there is a diff between the rendered files and the source files.  

if you have pre-commit installed you can run `pre-commit run` or `hack/enable-git-hooks.sh` to enable the validation hooks locally

File: src/functions/chk_file.sh [hunk 2]
```diff
@@ -81,10 +84,14 @@ chk_file_list() {
...
-      echo_message "${CHK_FILE}\n" ",${CHK_FILE}\n" " filename='${CHK_FILE}'/>\n" " },\n"
+      LINE_ENDING=" },\n"
+      if [[ $CURRENT_LINE_NUMBER -eq $LAST_LINE_NUMBER ]]; then
+        LINE_ENDING=" }\n"
+      fi
+      echo_message "${CHK_FILE}\n" ",${CHK_FILE}\n" " filename='${CHK_FILE}'/>\n" "${LINE_ENDING}"
     fi
 
-  done < "${CHK_FILE_LIST}"
+  done <"${CHK_FILE_LIST}"
```
Comment (slimm609) [inline src/functions/chk_file.sh:94]: I think shfmt is failing on this space removal

Comment (slimm609) [reply src/functions/chk_file.sh:94]: https://github.com/mvdan/sh

Comment (ysmaoui) [reply src/functions/chk_file.sh]: weird,  shfmt in vscode formatted by removing that space..

committed again
now the hooks passed locally ( thanks for the tips )

**Pull request merged**